api/Service/PictureService.cs

```diff
@@ -0,0 +1,70 @@
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using YonderfulApi.Models;
+using System.Drawing;
+using YonderfulApi.Data;
+using Microsoft.EntityFrameworkCore;
+
+namespace YonderfulApi.Service 
+{
+  public class PictureService : IPictureService
+  {
+    private readonly DataContext _context;
+    public PictureService(DataContext context) {
+        _context = context;
+    }
+    public async Task<Picture> GetPicture(int pictureId)
+    {
+      var picture = await _context.Pictures.FindAsync(pictureId);
+      return picture;
+    }
+
+    public async Task<IList<Picture>> GetPictureList()
+    {
+        var pictureList = await _context.Pictures.ToListAsync();
+        return pictureList;
+    }
+
+    public async Task<Picture> GetPictureByNameFormatContent(string name, string fileType, byte[] content)
+    {
+      var pictureList = await GetPictureList();
+      foreach(Picture picture in pictureList) {
+        if(picture.Name.ToLower() == name.ToLower() && picture.FileType.ToLower() == fileType.ToLower() && picture.Content.Equals(content)) {
+          return picture;
+        }
+      }
+      return null;
+    }
+    public async Task<Picture> PostPicture(string name, string fileType, byte[] content)
+    {
+
+        // _pictureService.GetPictureByNameFormatContent(fileName, fileType, content).Id;
+
+        var newPicture = new Picture 
```

Comment:
This mapping can be done in the Controller, so the service gets a `Picture` object as a parameter.